CMakeLists.txt

```diff
@@ -59,268 +58,251 @@ include (GitRepo)
-
-# sndfile
-pkg_check_modules(SNDFILE REQUIRED "sndfile >= 1.0.25")
-add_includes_ldflags ("${SNDFILE_LDFLAGS}" "${SNDFILE_INCLUDE_DIRS}")
-
-## Additional features ##
+pkg_check_modules(ExternalModules REQUIRED IMPORTED_TARGET ${pkg-module-spec})
+find_package(ZLIB REQUIRED)
+find_package(Threads REQUIRED)
+if (APPLE)
+    pkg_check_modules(MacIntegration IMPORTED_TARGET "gtk-mac-integration")
+elseif (NOT WIN32)
+    find_package(Backtrace REQUIRED)
+endif ()
 
-# CppUnit
-option (ENABLE_CPPUNIT "Build CppUnit test instead of xournalpp application" OFF)
+add_library(external_modules INTERFACE)
+target_link_libraries(external_modules INTERFACE
+        PkgConfig::ExternalModules
+        $<$<TARGET_EXISTS:PkgConfig::MacIntegration>:PkgConfig::MacIntegration>
+        $<$<TARGET_EXISTS:PkgConfig::Lua>:PkgConfig::Lua>
+        ZLIB::ZLIB
+        Threads::Threads
+        xoj::stacktrace
+        $<$<PLATFORM_ID:Linux>:-lX11 -lXi>
+
+        )
+target_compile_definitions(external_modules INTERFACE
+        GLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_32
+        GDK_VERSION_MIN_REQUIRED=GDK_VERSION_3_18
+        $<$<TARGET_EXISTS:PkgConfig::MacIntegration>:MAC_INTEGRATION>
+        $<$<TARGET_EXISTS:PkgConfig::Lua>:ENABLE_PLUGINS>
+        )
+target_compile_options(external_modules INTERFACE
+        $<$<AND:$<PLATFORM_ID:Windows>,$<CXX_COMPILER_ID:GNU>>:-rdynamic>
+        )
+add_library(xoj::external_modules ALIAS external_modules)
+
+# CppUnit Todo move to test
+option(ENABLE_CPPUNIT "Build CppUnit test instead of xournalpp application" OFF)
 if (ENABLE_CPPUNIT)
-	pkg_check_modules(CppUnit REQUIRED "cppunit >= 1.12-0")
-	enable_testing()
 endif (ENABLE_CPPUNIT)
 
-# Mac integration
-pkg_check_modules (MacIntegration "gtk-mac-integration")
-if (MacIntegration_FOUND)
-	message("Enable Mac integration")
-	add_includes_ldflags ("${MacIntegration_LDFLAGS}" "${MacIntegration_INCLUDE_DIRS}")
-	add_definitions (-DMAC_INTEGRATION)
-endif ()
-
-# Plugins / scripting
-pkg_check_modules (Lua "lua5.3")
-if (NOT Lua_FOUND)
-	pkg_check_modules (Lua "lua >= 5.3")
-endif()
-
 if (Lua_FOUND)
-	message("Enable Xournal++ Plugins")
-	add_includes_ldflags("${Lua_LDFLAGS}" "${Lua_INCLUDE_DIRS}")
-	set(ENABLE_PLUGINS "true")
+    message("Enable Xournal++ Plugins")
+    set(ENABLE_PLUGINS "true")
 endif ()
 
 unset(add_includes_ldflags)
 
-#
+#[[#
 # DO NOT INCLUDE LIBRARIES WITH pkg_check_modules AFTER THIS LINE!!!
 #
-if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-	# Fix linker flag handling for MacOS builds where frameworks are separated from their flag
-	string(REPLACE "-framework;" "-framework " xournalpp_LDFLAGS "${xournalpp_LDFLAGS}")
-endif()
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    # Fix linker flag handling for MacOS builds where frameworks are separated from their flag
+    string(REPLACE "-framework;" "-framework " xournalpp_external_deps "${xournalpp_external_deps}")
+endif ()]]
```

Comment:
Are you sure this fix is no longer required?

Reply:
Not sure, but I think thats the job of cmake.

Reply:
I had to include this fix way back because CMake did not do this. We have to check the pipeline for this and do some test runs.

Reply:
Nah I think, this is obsolete after my fix / it's our fault, when this is necessary.

Reply:
Then please remove it completely.